### PR TITLE
Fix in object detection task to support train and predict on cpu node.

### PR DESCRIPTION
**Pull request by mak-454**
*Issue #, if available:*

*Description of changes:*
PR provides a fix in object detection task when ngpus_per_trial is set to 0 for detector. 
**Error observed**
_ind, prob, loc = detector.predict(image_path)_ 
>   File "/usr/local/lib/python3.6/site-packages/autogluon/task/object_detection/detector.py", line 171, in predict
>     net.collect_params().reset_ctx(ctx=mx.cpu())
>   File "/usr/local/lib/python3.6/site-packages/mxnet/gluon/parameter.py", line 929, in reset_ctx
>     i.reset_ctx(ctx)
>   File "/usr/local/lib/python3.6/site-packages/mxnet/gluon/parameter.py", line 493, in reset_ctx
>     "has not been initialized."%self.name)
> ValueError: Cannot reset context for Parameter 'mobilenet0_conv0_weight' because it has not been initialized.

**Fix**
To set the cpu context if num of gpus == 0

**Verification script**
Verified by passing _ngpus_per_trial=0_ to the _detector_
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Innixma — CHANGES_REQUESTED** [review]
Thanks for the contribution!

@zhreshold Could you take a look and approve if all is good?

I am blocking merge for now since we are in code-freeze to enable a major modularization PR merge (#694). Once this is done (estimated ~1-2 weeks), I will unblock and merge given approvals have been made.

**Review by zhreshold — APPROVED** [review]
lgtm!

**Review by daveharmon — APPROVED** [review]

File: autogluon/task/object_detection/object_detection.py
Comment (zhreshold) [inline autogluon/task/object_detection/object_detection.py:299]: @mak-454 After modularization, this file has been moved to `vision/src/autogluon/vision/object_detection/object_detection.py`, can you update the code? I can help review the change again!

**Review by zhreshold — CHANGES_REQUESTED** [review]

File: autogluon/task/object_detection/object_detection.py
Comment (mak-454) [reply autogluon/task/object_detection/object_detection.py:299]: @zhreshold sure will do.

Comment (mak-454) [reply autogluon/task/object_detection/object_detection.py]: @zhreshold please review.

**Comment by szha** [comment]
Job PR-697-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-697/2/index.html

**Comment by szha** [comment]
Job PR-697-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-697/3/index.html

**Review by Innixma — APPROVED** [review]
Approving, @zhreshold feel free to merge.

**Review by zhreshold — APPROVED** [review]

**Pull request merged**
